web/main.js

```diff
@@ -116,33 +124,58 @@ let send = async f => {
 let receive = e => {
 	if (!receiving) {
 		receiving = JSON.parse(new TextDecoder('utf8').decode(e.data));
-		receiving.data = new Uint8Array(receiving.size);
+		// Maybe there is a better way to identify a single file transfer.
+		receiving.id = document.getElementById("magiccode").value + '-' + Date.now().toString(16);
```

Comment:
This is almost certainly fine here. We could even add a random number at the end for good measure.

It would be safest if the worker allocated the id, which can synchronously keep a counter, or at least check for clashes. Maybe a POST to /download/ returns an id? But then we'd have to wait for its response and queue any chunks we receive before we have an id... yeah let's keep it here.

Reply:
Changed to random.